collections/src/main/scala/strawman/collection/ArrayOps.scala

```diff
@@ -54,11 +69,158 @@ class ArrayOps[A](val xs: Array[A])
 
   def zip[B: ClassTag](xs: Iterable[B]): Array[(A, B)] = fromTaggedIterable(View.Zip(toIterable, xs))
 
+  def append[B >: A : ClassTag](x: B): Array[B] = fromTaggedIterable(View.Append(toIterable, x))
+  @`inline` final def :+ [B >: A : ClassTag](x: B): Array[B] = append(x)
+  def prepend[B >: A : ClassTag](x: B): Array[B] = fromTaggedIterable(View.Prepend(x, toIterable))
+  @`inline` final def +: [B >: A : ClassTag](x: B): Array[B] = prepend(x)
 }
 
-case class ArrayView[A](xs: Array[A]) extends IndexedView[A] {
-  def length = xs.length
-  @throws[ArrayIndexOutOfBoundsException]
-  def apply(n: Int) = xs(n)
+abstract class ArrayView[A] extends IndexedView[A] {
+  //def length = xs.length
+  //@throws[ArrayIndexOutOfBoundsException]
+  //def apply(n: Int) = xs(n)
   override def className = "ArrayView"
 }
+
+object ArrayView {
+  // This is reused for all calls to empty.
+  private val EmptyArrayView  = new ofRef[AnyRef](new Array[AnyRef](0))
+  def empty[T <: AnyRef]: ArrayView[T] = EmptyArrayView.asInstanceOf[ArrayView[T]]
+
+  @`inline` def apply[T](x: Array[T]): ArrayView[T] = make(x)
+
+  // If make is called explicitly we use whatever we're given, even if it's
+  // empty.  This may be unnecessary (if ArrayView is to honor the collections
+  // contract all empty ones must be equal, so discriminating based on the reference
+  // equality of an empty array should not come up) but we may as well be
+  // conservative since wrapRefArray contributes most of the unnecessary allocations.
+  def make[T](x: AnyRef): ArrayView[T] = (x match {
```

Comment:
Why is the type of `x` `AnyRef` rather than `Array[_]` (for instance)? Do we want this method to be public?

Reply:
It's currently done with `AnyRef` but it should work with `Array[_]` (which also erases to `Object`). Since this is the only way of creating an `ArrayView` that takes primitive arrays into account, it needs to be public.